### ISSUE TITLE
logrotate on xrdp-sesman causes window manager problems

### ISSUE THREAD

**Issue by TomTrenker** [neutrinolabs/xrdp#1291]
If an existing rdp session is active or disconnected when logrotate sends a HUP to 
rotate xrdp-sesman.log, future sessions fail to bring up the window manager after logoff.

While the HUP on xrdp-sesman doesn't affect the existing session, and the user can 
reconnect to the existing session, if the user logs off from their session then any 
following connections have no window manager.  The user is presented with a session
with an open xterm and nothing else.  xrdp-sesman.log shows:
```
[20190209-21:19:24] [INFO ] A connection received from 127.0.0.1 port 62873
[20190209-21:19:25] [INFO ] ++ created session (access granted): username *******, ip 127.0.0.1:65119 - socket: 12
[20190209-21:19:25] [INFO ] starting Xorg session...
[20190209-21:19:25] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5911)
[20190209-21:19:25] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:0)
[20190209-21:19:25] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5912)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:0)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5913)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:0)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5914)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:0)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5915)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:0)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5916)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6016)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6216)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5917)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6017)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6217)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5918)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6018)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6218)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5919)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6019)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6219)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:5928)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6028)
[20190209-21:19:26] [DEBUG] Closed socket 13 (AF_INET 0.0.0.0:6228)
[20190209-21:19:26] [DEBUG] Closed socket 10 (AF_INET 127.0.0.1:3350)
[20190209-21:19:26] [INFO ] calling auth_start_session from pid 146374
[20190209-21:19:26] [DEBUG] Closed socket 9 (AF_INET 127.0.0.1:3350)
[20190209-21:19:26] [DEBUG] Closed socket 10 (AF_INET 127.0.0.1:3350)
[20190209-21:19:26] [INFO ] Xorg :28 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp -logfile .xorgxrdp.%s.log  
[20190209-21:19:26] [CORE ] error starting default wm for user ******* - pid 146375
[20190209-21:19:26] [CORE ] waiting for window manager (pid 146375) to exit
[20190209-21:19:27] [DEBUG] errno: 2, description: No such file or directory
[20190209-21:19:27] [DEBUG] execlp3 parameter list:
[20190209-21:19:27] [DEBUG]         argv[0] = startwm.sh
[20190209-21:19:27] [DEBUG]         argv[1] = startwm.sh
```
- service xrdp.sh restart is required to correct the problem.

- information about my xrdp installation:
```
qgccws1 (root):#/etc/init.d> /usr/local/sbin/xrdp -v
xrdp 0.9.9
  A Remote Desktop Protocol Server.
  Copyright (C) 2004-2018 Jay Sorg, Neutrino Labs, and all contributors.
  See https://github.com/neutrinolabs/xrdp for more information.

  Configure options:
      

  Compiled with OpenSSL 1.0.1e-fips 11 Feb 2013

qgccws1 (root):#/etc/init.d> uname -a
Linux ******* 2.6.32-642.13.1.el6.x86_64 #1 SMP Tue Jan 10 09:48:48 PST 2017 x86_64 x86_64 x86_64 GNU/Linux


qgccws1 (root):#/etc> cat /etc/system-release
Oracle Linux Server release 6.8
```
- any suggestions?
- I can remove xrdp from logrotate.d and I won't be affected by this problem, but I'd prefer a better solution.

Thanks - Tom

**Comment by metalefty** [comment]
I've never used logrotate for xrdp so I didn't notice that. Thanks for reporting. The issue needs to be addressed.

**Comment by metalefty** [comment]
This issue is identical to #1315 and fixed at #1318. Can you test?
